src/DataMapper/ActivityMapper.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GroupLife\Core\DataMapper;
+
+use GroupLife\Core\Activity;
+
+class ActivityMapper
+{
+    private $connection;
+    private $scheduleMapper;
+    private $leaderMapper;
+
+    public function __construct(\Doctrine\DBAL\Connection $connection)
+    {
+        $this->connection = $connection;
+        $this->scheduleMapper = new ScheduleMapper($connection);
+        $this->leaderMapper = new LeaderMapper($connection);
+    }
+
+    /**
+     * @param Activity $activity
+     * @throws \Doctrine\DBAL\Exception
+     */
+    public function insert(Activity $activity)
+    {
+        $this->scheduleMapper->insert($activity->getData()->schedule);
+        $this->leaderMapper->insert($activity->getData()->leader);
```

Comment:
What if a leader is already persists? In real world it will be the case.

Reply:
Fixed